From bfabf8f8d966ab8305212b9730db8fc5eb5e1094 Mon Sep 17 00:00:00 2001
From: Timo Teras <timo.teras@iki.fi>
Date: Wed, 15 Jul 2009 08:38:30 +0300
Subject: [PATCH] pkg: fix package scanner to read whole archive

otherwise we don't get chechksum anymore after the changes to
support partial gzip checksumming.
---
 src/package.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/package.c b/src/package.c
index bb37317..6c050cf 100644
--- a/src/package.c
+++ b/src/package.c
@@ -372,7 +372,7 @@ static int read_info_entry(void *ctx, const struct apk_file_info *ae,
 		if (strcmp(ae->name, ".INSTALL") == 0) {
 			apk_warning("Package '%s-%s' contains deprecated .INSTALL",
 				    pkg->name->name, pkg->version);
-			return 1;
+			return 0;
 		}
 	} else if (strncmp(ae->name, "var/db/apk/", 11) == 0) {
 		/* APK 1.0 format */
@@ -407,7 +407,7 @@ static int read_info_entry(void *ctx, const struct apk_file_info *ae,
 			ri->has_install = 1;
 	} else if (ri->version == 2) {
 		/* All metdata of version 2.x package handled */
-		return 1;
+		return 0;
 	} else {
 		/* Version 1.x packages do not contain installed size
 		 * in metadata, so we calculate it here */
-- 
1.6.3.3

